django/thunderstore/social/api/experimental/views/mod_manager.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.social.api.experimental.views.complete_login import (
+    get_or_create_auth_user,
+)
+from thunderstore.social.permissions import OauthSharedSecretPermission
+from thunderstore.social.providers import UserInfoSchema
+
+User = get_user_model()
+
+
+class MmLoginRequestBody(serializers.Serializer):
+    jwt = serializers.CharField(label="Authorization token")
+
+
+class MmLoginResponseBody(serializers.Serializer):
+    session_id = serializers.CharField()
+    username = serializers.CharField()
+
+
+class OverwolfProfileSchema(BaseModel):
+    """
+    https://overwolf.github.io/api/profile#get-user-profile-via-token
+    """
+
+    avatar: str
+    nickname: str
+    username: str
+
+
+class ModManagerLoginApiView(APIView):
+    """
+    Login user with information from Overwolf API using the received JWT
+    """
+
+    permission_classes: Sequence[Type[BasePermission]] = [OauthSharedSecretPermission]
+
+    @swagger_auto_schema(
+        operation_id="experimental.auth.mod_manager",
+        request_body=MmLoginRequestBody,
+        responses={200: MmLoginResponseBody()},
+    )
+    def post(self, request: Request) -> HttpResponse:
+        request_data = MmLoginRequestBody(data=request.data)
+        request_data.is_valid(raise_exception=True)
+        jwt: str = request_data.validated_data["jwt"]
+
+        verify_overwolf_jwt(jwt)
+        ow_profile = get_overwolf_user_profile(jwt)
+        user_info = get_user_info(ow_profile)
+        user = get_or_create_auth_user(user_info)
+        login(request, user, "django.contrib.auth.backends.ModelBackend")
+
+        if not request.session.session_key:
+            request.session.create()
+
+        return Response(
+            {
+                "session_id": request.session.session_key,
+                "username": user.username,
+            }
+        )
+
+
+def query_overwolf_jwt_api(path: str, jwt: str) -> requests.Response:
+    ow_id = settings.OVERWOLF_CLIENT_ID
+    url = f"https://accounts.overwolf.com/tokens/short-lived/{path}?extensionId={ow_id}"
+    headers = {"Authorization": f"Bearer {jwt}"}
+    return requests.get(url, headers=headers)
+
+
+def verify_overwolf_jwt(jwt: str) -> None:
+    response = query_overwolf_jwt_api("verify", jwt)
+
+    if response.status_code != 200:
+        raise AuthenticationFailed("Invalid or expired JWT")
+
+
+def get_overwolf_user_profile(jwt: str) -> OverwolfProfileSchema:
+    response = query_overwolf_jwt_api("users/profile", jwt)
+    return OverwolfProfileSchema.parse_obj(response.json())
+
+
+def get_user_info(profile: OverwolfProfileSchema) -> UserInfoSchema:
+    """
+    Overwolf doesn't provide a way to query user's email address, so
+    make one up, like we do for service accounts.
+    """
+    return UserInfoSchema.parse_obj(
+        {
+            "email": f"{ulid2.generate_ulid_as_uuid().hex}.mm@thunderstore.io",
```

Comment:
I'd prefer the suffix would end with something like `.autogenerated@thunderstore.io` to make filtering for them unambiguous.

Reply:
`.mm` was chosen to match `.sa` already used by service accounts.

Reply:
Yeah I was just thinking we might use that pattern outside of the mod manager context too, but I suppose it doesn't matter as long as it's consistent